main.tf

```diff
@@ -303,3 +305,175 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
     }
   }
 }
+
+resource "aws_cloudfront_distribution" "cloudfront_staging_distribution" {
+  count               = var.enable_cloudfront_staging ? 1 : 0
+  comment             = "${var.comment}. Staging associated with resource ID ${random_id.this.hex}"
+  default_root_object = var.default_root_object
+  enabled             = var.enable_cloudfront_staging
+  http_version        = "http2"
```

Comment:
looking at the doc http3 is not supported, preventing the staging distribution by being deployed. Should we specify anywhere this information? Should we already hardcode http2 for the production distribution?

Reply:
I wrote the following about http2 requirement in the readme file. 

> Also, as soon as you want to use the staging distribution feature, `http2` is required as the `http_version`.

Migrating to http2 could be done when migrating to the new module version IMO.